src/openqaoa-core/tests/test_bpsp.py

```diff
@@ -53,69 +45,77 @@ def test_bpsp_terms_weights_constant(self):
         self.assertEqual(gr_weights, bpsp_prob_qubo.weights)
         self.assertEqual(0.0, bpsp_prob_qubo.constant)
 
-    def test_car_sequence_raises(self):
+    def test_bpsp_car_sequence_raises(self):
         """
         Test if ValueErrors are raised for invalid car sequences.
 
         Certain car sequences are expected to be invalid based on their content.
         This method checks if setting such sequences raises a ValueError.
         """
+        bpsp = BPSP(np.array([0, 1, 0, 1]))
         with self.assertRaises(ValueError):
-            self.bp.car_sequence = np.array([0, 1, 0])
+            bpsp.car_sequence = np.array([0, 1, 0])
         with self.assertRaises(ValueError):
-            self.bp.car_sequence = np.array([0, 0, 1, 2, 2, 3, 3, 3])
+            bpsp.car_sequence = np.array([0, 0, 1, 2, 2, 3, 3, 3])
 
-    def test_random_instance(self):
+    def test_bpsp_random_instance(self):
         """
         Test the generation of a random BPSP instance.
 
         Validates that the generated random BPSP instance has one of the
         expected car sequences.
         """
-        instance = BPSP.random_instance(num_cars=2)
-        self.assertTrue(all(np.array([0, 1, 0, 1]) == instance.car_sequence) or 
-                        all(np.array([1, 1, 0, 0]) == instance.car_sequence) or 
-                        all(np.array([0, 0, 1, 1]) == instance.car_sequence) or 
-                        all(np.array([1, 0, 1, 0]) == instance.car_sequence))
+        bpsp = BPSP.random_instance(num_cars=2, seed=1234)
+        self.assertTrue(all(np.array([0, 1, 0, 1]) == bpsp.car_sequence) or 
+                        all(np.array([1, 1, 0, 0]) == bpsp.car_sequence) or 
+                        all(np.array([0, 0, 1, 1]) == bpsp.car_sequence) or 
+                        all(np.array([1, 0, 1, 0]) == bpsp.car_sequence))
 
-    def test_car_pos(self):
+    def test_bpsp_car_pos(self):
         """Test the retrieval of car positions."""
-        self.assertEqual(self.bp.car_positions, {0: (0, 2), 1: (1, 3)})
+        bpsp = BPSP(np.array([0, 1, 0, 1]))
+        self.assertEqual(bpsp.car_positions, {0: (0, 2), 1: (1, 3)})
 
-    def test_graph(self):
+    def test_bpsp_graph(self):
         """Test the generation of a graph representation of the BPSP instance."""
-        self.assertEqual(len(self.bp.bpsp_graph.nodes), 2)
+        bpsp = BPSP.random_instance(num_cars=2, seed=1234)
+        self.assertEqual(len(bpsp.bpsp_graph.nodes), 2)
 
-    def test_docplex_bpsp_model(self):
+    def test_bpsp_docplex_bpsp_model(self):
         """Test if the docplex model representation of BPSP is generated."""
-        self.assertIsNotNone(self.bp.docplex_bpsp_model)
+        bpsp = BPSP.random_instance(num_cars=2, seed=1234)
+        self.assertIsNotNone(bpsp.docplex_bpsp_model)
```

Comment:
Check the attributes of your docplex model if possible

Reply:
I'm now checking the number of variables, number of constraints, name of the variables, checks if the variables are indeed binary, whether the model has an objective function, and if the objective mode is set to minimize.